ledger/metrics/src/main/scala/com/daml/metrics/InstrumentedSource.scala

```diff
@@ -3,42 +3,37 @@
 
 package com.daml.metrics
 
-import akka.Done
-import akka.stream.scaladsl.{Source, SourceQueueWithComplete}
-import akka.stream.{Materializer, OverflowStrategy, QueueOfferResult}
+import akka.stream.scaladsl.Source
+import akka.stream.{BoundedSourceQueue, Materializer, OverflowStrategy, QueueOfferResult}
 import com.codahale.metrics.{Counter, Timer}
-import com.daml.dec.DirectExecutionContext
-
-import scala.concurrent.Future
 
 object InstrumentedSource {
 
-  final class QueueWithComplete[T](
-      delegate: SourceQueueWithComplete[(Timer.Context, T)],
+  final class InstrumentedBoundedSourceQueue[T](
+      delegate: BoundedSourceQueue[(Timer.Context, T)],
+      bufferSize: Int,
+      capacityCounter: Counter,
       lengthCounter: Counter,
       delayTimer: Timer,
-  ) extends SourceQueueWithComplete[T] {
+  ) extends BoundedSourceQueue[T] {
 
-    override def complete(): Unit = delegate.complete()
+    override def complete(): Unit = {
+      delegate.complete()
+      capacityCounter.dec(bufferSize.toLong)
```

Comment:
Is the idea here to basically reset to 0? If so, is there no API to directly set the expected value?

Reply:
I think so, no :(